react/src/Analyses/components/AnalysisNotes.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { Divider, Popover, Typography } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { useAnalysisQuery } from "../../hooks";
+import { Analysis, Dataset } from "../../typings";
+
+interface AnalysisNotesProps {
+    analysis: Analysis;
+    anchorEl: HTMLButtonElement | null;
+    open: boolean;
+    onClose: () => void;
+}
+
+interface DatasetWithNotes extends Dataset {
+    participant_notes: string;
+    dataset_notes: string;
+}
+
+const useStyles = makeStyles(theme => ({
+    paper: {
+        padding: theme.spacing(2),
+    },
+    divider: {
+        margin: theme.spacing(1, 0),
+    },
+    comment: {
+        fontWeight: 500,
+    },
+}));
+
+export default function AnalysisNotes(props: AnalysisNotesProps) {
+    const classes = useStyles();
+    const { data: analysisQueryResults } = useAnalysisQuery(props.analysis.analysis_id);
+    const analysisWithNotes = (
+        (analysisQueryResults && (analysisQueryResults.datasets as DatasetWithNotes[])) ||
+        []
+    ).filter(dataset => dataset.notes || dataset.participant_notes);
+
+    return (
+        <Popover
+            PaperProps={{ className: classes.paper }}
+            open={props.open}
+            anchorEl={props.anchorEl}
+            onClose={props.onClose}
+            anchorOrigin={{
+                vertical: "bottom",
+                horizontal: "left",
+            }}
+        >
+            {analysisWithNotes.length > 0 ? (
+                <>
+                    <Typography variant="h6">Notes</Typography>
+                    {analysisWithNotes.map(dataset => (
+                        <>
+                            <Divider className={classes.divider} />
+                            <Typography className={classes.comment}>
+                                Comment by {dataset.updated_by} at {dataset.updated}
+                            </Typography>
+                            <Typography variant="subtitle1">
+                                Dataset {dataset.dataset_id}
```

Comment:
The primary key is actually not very helpful to a person. We have done something like `family_codename/participant_codename/dataset_type` for the dataset delete button to be more identifying.